validation/main.cpp

```diff
@@ -116,16 +118,35 @@ int main(int argc, char *argv[]) {
             return EXIT_FAILURE;
         }
     }
+
+    QStringList commandList = {"time_settings 0 1 0"};
+    commandList << parser.values(gtpCommandOption);
+
+    auto engines = QVector<engine_t>({
+        {"./leelaz", optsList[0], netList[0], commandList},
+        {"./leelaz", optsList[1], netList[1], commandList}});
+
+    auto engine_idx = 0;
+    auto pos_args = parser.positionalArguments();
+    while (!pos_args.isEmpty()) {
+        if (engine_idx >= 2) {
+            parser.showHelp();
+        }
+        engines[engine_idx].binary = pos_args[0];
+        parser.process(pos_args);
+        engines[engine_idx].commands << parser.values(gtpCommandOption);
+        pos_args = parser.positionalArguments();
+        engine_idx++;
+    }
+
```

Comment:
Does this while loop works like we want?

Reply:
From my limited usage of the changes it appears to work how I expect. Have you got any particular concerns? I might have misunderstood something :) 
I was surprised at first that I didn't have to remove/pop the binary from `pos_args` but `QCommandLineParser::process()` expects the string list of arguments to be from the command line and therefore ignores the first item as "Usually arguments().at(0) is the program name..." http://doc.qt.io/qt-5/qcoreapplication.html#arguments. I have only got Windows to test on though.
I could add some comment(s) if that helps?